code/modules/mapping/mapping_helpers.dm

```diff
@@ -851,3 +851,96 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 	var/turf/open/floor/floor = get_turf(src)
 	floor.burn_tile()
 	qdel(src)
+
+
+///Causes a room to be trashed and graffitied if the revs won in the last round
+/obj/effect/mapping_helpers/revolution_trash
+	name = "revolution room trasher"
+	late = TRUE
+	icon_state = "revtrasher"
+	var/static/list/trash_talk = list(
+		"amyjon",
+		"antilizard",
+		"body",
+		"cyka",
+		"danger",
+		"electricdanger",
+		"face",
+		"guy",
+		"matt",
+		"peace",
+		"poseur tag",
+		"prolizard",
+		"radiation",
+		"revolution",
+		"shotgun",
+		"skull",
+		"splatter",
+		"star",
+		"stickman",
+		"toilet",
+		"toolbox",
+		"uboa",
+	)
+
+/obj/effect/mapping_helpers/revolution_trash/Initialize(mapload)
+	.=..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/mapping_helpers/revolution_trash/LateInitialize()
+	if(!HAS_TRAIT(SSstation, STATION_TRAIT_REVOLUTIONARY_TRASHING))
+		qdel(src) //We do nothing if our station trait isnt here
+		return
+
+	var/area/our_area = get_area(src)
+
+	for(var/atom/current_thing in our_area.contents)
```

Comment:
`as anything`

Also, instead of `.contents` (which is *extremely* slow for areas), use the proc we made for this. I think it's `get_area_contents()`? Lemon would know